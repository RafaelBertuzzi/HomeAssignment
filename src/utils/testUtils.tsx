import { ReactNode } from 'react';
import renderer from 'react-test-renderer';

import { LayoutProvider } from '@components/LayoutProvider';
import { render } from '@testing-library/react-native';

export const rendererWithProviders = (children: ReactNode) => {
  return renderer.create(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};

export const renderWithProviders = (children: ReactNode) => {
  return render(<LayoutProvider>{children}</LayoutProvider>).toJSON();
};
