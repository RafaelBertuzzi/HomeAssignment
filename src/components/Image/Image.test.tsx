import { cleanup } from '@testing-library/react-native';
import { rendererWithProviders } from '@utils/testUtils';

import { Image } from '.';

describe('Component: Image', () => {
  afterEach(cleanup);

  it('should render image', () => {
    const json = rendererWithProviders(
      <Image
        alt="search"
        size={6}
        source={require('../../assets/icons/search.png')}
      />,
    );
    expect(json).toMatchSnapshot();
  });
});
