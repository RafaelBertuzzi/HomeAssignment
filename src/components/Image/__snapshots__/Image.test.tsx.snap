// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Image should render image 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <Image
    accessibilityLabel="search"
    alt="search"
    dataSet={{}}
    onError={[Function]}
    source={
      {
        "testUri": "../../../src/assets/icons/search.png",
      }
    }
    style={
      {
        "height": 24,
        "maxWidth": "100%",
        "width": 24,
      }
    }
  />
</RNCSafeAreaProvider>
`;
